includes/Admin/Admin_AJAX.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * Class WordPress\Plugin_Check\Admin\Admin_AJAX
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\Admin;
+
+/**
+ * Class to handle the Admin AJAX requests.
+ *
+ * @since n.e.x.t
+ */
+class Admin_AJAX {
+
+	/**
+	 * Nonce key.
+	 *
+	 * @since n.e.x.t
+	 * @var string
+	 */
+	private $nonce_key;
+
+	/**
+	 * Sets the nonce key for verification.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $nonce_key Nonce key.
+	 */
+	public function __construct( $nonce_key ) {
+		$this->nonce_key = $nonce_key;
+	}
+
+	/**
+	 * Registers WordPress hooks for the Admin AJAX.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function add_hooks() {
+		add_action( 'wp_ajax_plugin_check_run_checks', array( $this, 'run_checks' ) );
+	}
+
+	/**
+	 * Creates and returns the nonce.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function get_nonce() {
+		return wp_create_nonce( $this->nonce_key );
+	}
+
+	/**
+	 * Run checks.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function run_checks() {
+		$nonce = filter_input( INPUT_POST, 'nonce', FILTER_SANITIZE_STRING );
+
+		if ( ! wp_verify_nonce( $nonce, $this->nonce_key ) ) {
+			wp_send_json_error();
```

Comment:
We should pass an actual error message here. Maybe pass a `new WP_Error( ... )` to `wp_send_json_error()` as that function will automatically put the data into the error response in a consistent format.

We also need to send an error status code. Probably a `403` works well here.